packages/core/src/modules/dids/methods/jwk/DidJwk.ts

```diff
@@ -0,0 +1,55 @@
+import type { Jwk } from '../../../../crypto'
+
+import { Key } from '../../../../crypto/Key'
+import { JsonEncoder } from '../../../../utils'
+import { parseDid } from '../../domain/parse'
+
+import { getDidJwkDocument } from './didJwkDidDocument'
+
+export class DidJwk {
+  public readonly did: string
+
+  private constructor(did: string) {
+    this.did = did
+  }
+
+  public get forEncrypting() {
```

Comment:
Naming seems a bit vague to me here. Maybe `canEncrypt()` and `canSign()` or something similar?

Reply:
Yeah maybe `allowsEncrypting` and `allowsSigning`? Think we should differentiate between can and allowed to

Reply:
YES! Even better!